openpype/pipeline/publish/lib.py

```diff
@@ -305,8 +306,15 @@ def publish_plugins_discover(paths=None):
             module.__file__ = abspath
 
             try:
-                with open(abspath, "rb") as f:
-                    six.exec_(f.read(), module.__dict__)
+                if six.PY3:
```

Comment:
This should maybe use `import_filepath` from `openpype.lib`.

Reply:
oh yeah, we should be able to replace this whole code block by it, thank you! let me give it a go

Reply:
Done! I had to modify `openpype.lib.import_filepath` so it assigns __file__ on the module because I was getting the following error:
```
Traceback (most recent call last):
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/tools/publisher/window.py", line 456, in _on_show_timer
    self.reset()
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/tools/publisher/window.py", line 462, in reset
    self._controller.reset()
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/tools/publisher/control.py", line 1762, in reset
    self._reset_plugins()
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/tools/publisher/control.py", line 1780, in _reset_plugins
    self._create_context.reset_plugins()
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/pipeline/create/context.py", line 1620, in reset_plugins
    self._reset_publish_plugins(discover_publish_plugins)
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/pipeline/create/context.py", line 1638, in _reset_publish_plugins
    discover_result = publish_plugins_discover()
  File "/home/farrizabalaga/dev/OpenPype/build/exe.linux-x86_64-3.9/openpype/pipeline/publish/lib.py", line 328, in publish_plugins_discover
    plugin.__module__ = module.__file__
AttributeError: module 'collect_current_user' has no attribute '__file__'
```

Reply:
Changes done here https://github.com/ynput/OpenPype/pull/4498/commits/9dec1420b134618cf34959d19c5b04d4956bb2a6